### PR TITLE
feat(BUGS-2826): recovery tool

### PR DESCRIPTION
**Pull request by gponomarev-lykke**

### CODE REVIEW

File: MarginTrading.AccountsManagement.RecoveryTool/LogParsers/AccountsManagementLogParser.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace MarginTrading.AccountsManagement.RecoveryTool.LogParsers;
+
+public class AccountsManagementLogParser
+{
+    public List<DomainEvent> Parse(string log)
+    {
+        var result = new List<DomainEvent>();
+            
+        var margin = Create("UpdateBalanceCommandsHandler:");
```
Comment (tarurar) [inline MarginTrading.AccountsManagement.RecoveryTool/LogParsers/AccountsManagementLogParser.cs:18]: why is it "margin"?

Comment (tarurar) [reply MarginTrading.AccountsManagement.RecoveryTool/LogParsers/AccountsManagementLogParser.cs:18]: just curious.

**Review by tarurar — DISMISSED** [review]

File: MarginTrading.AccountsManagement.RecoveryTool/LogParsers/AccountsManagementLogParser.cs (hunk 1 above)
Comment (gponomarev-lykke) [reply MarginTrading.AccountsManagement.RecoveryTool/LogParsers/AccountsManagementLogParser.cs:18]: Oh, I think it's just copy-paste from the activities recovery tool. Will fix

File: MarginTrading.AccountsManagement.RecoveryTool/Services/Publisher.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+namespace MarginTrading.AccountsManagement.RecoveryTool.Services;
+
+public class Publisher : IDisposable
+{
+    private IConnection _connection;
+    private IModel _publishingChannel;
+
+    public Publisher(string rabbitMqConnectionString)
+    {
+        var factory = new ConnectionFactory {Uri = new Uri(rabbitMqConnectionString, UriKind.Absolute)};
```
Comment (tarurar) [inline MarginTrading.AccountsManagement.RecoveryTool/Services/Publisher.cs:20]: Let's use AutomaticRecoveryEnabled and set NetworkRecoveryInterval to, let's say 60 seconds.

File: MarginTrading.AccountsManagement.RecoveryTool/Services/Publisher.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public async Task Publish<T>(T command, string routingKey)
+    {
+        IBasicProperties properties = _publishingChannel.CreateBasicProperties();
+
+        var serializer = new MessagePackMessageSerializer<T>();
+        var message = serializer.Serialize(command);
+
+        _publishingChannel.BasicPublish("",
```
Comment (tarurar) [inline MarginTrading.AccountsManagement.RecoveryTool/Services/Publisher.cs:33]: Let's use mandatory option and handle messages which were not delivered.

File: MarginTrading.AccountsManagement.RecoveryTool/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Debug",
+      "System": "Information",
+      "Microsoft": "Information"
+    }
+  },
+  "ConnectionStrings": {
+    "db": "Server=13.69.102.164,1433;Initial Catalog=nova;Persist Security Info=False;User ID=novasql;Password=QAzP6b8rwwzJb#hc;MultipleActiveResultSets=False;Encrypt=False;Connection Timeout=90;",
```
Comment (tarurar) [inline MarginTrading.AccountsManagement.RecoveryTool/appsettings.json:10]: We can't publish sensitive data to git

**Review by tarurar — CHANGES_REQUESTED** [review]

**Review by tarurar — APPROVED** [review]

**Pull request merged**